x-pack/plugins/security_solution/server/lib/detection_engine/signals/enrichments/enrichment_by_type/user_risk.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { get, set, cloneDeep } from 'lodash';
+import { getUserRiskIndex } from '../../../../../../common/search_strategy/security_solution/risk_score/common';
+import { createSingleFieldMatchEnrichment } from '../create_single_field_match_enrichment';
+import type { CreateRiskEnrichment, GetIsRiskScoreAvailable } from '../types';
+
+export const getIsUserRiskScoreAvailable: GetIsRiskScoreAvailable = async ({
+  services,
+  spaceId,
+}) => {
+  const isUserRiskScoreIndexExist = await services.scopedClusterClient.asCurrentUser.indices.exists(
+    {
+      index: getUserRiskIndex(spaceId),
+    }
+  );
+
+  return isUserRiskScoreIndexExist;
+};
+
+export const createUserRiskEnrichments: CreateRiskEnrichment = async ({
+  services,
+  logger,
+  events,
+  spaceId,
+}) => {
+  return createSingleFieldMatchEnrichment({
+    name: 'User Risk',
+    index: [getUserRiskIndex(spaceId)],
+    services,
+    logger,
+    events,
+    mappingField: {
+      eventField: 'user.name',
+      enrichmentField: 'user.name',
+    },
+    createEnrichmentFunction: (enrichment) => (event) => {
+      const risk = get(enrichment, `_source.risk`);
+      if (!risk) {
+        return event;
+      }
+      const newEvent = cloneDeep(event);
```

Comment:
Are we cloning these due to race conditions with the parallelization?

Reply:
Really, we probably don't need to clone it here. But I want to be extra safe and not mutate original object